libs/api/ebpf_api.cpp

```diff
@@ -1948,3 +1948,71 @@ ebpf_object_close(_In_ _Post_invalid_ struct bpf_object* object)
     _remove_ebpf_object_from_globals(object);
     _clean_up_ebpf_object(object);
 }
+
+static ebpf_result_t
+_get_fd_by_id(ebpf_operation_id_t operation, ebpf_id_t id, _Out_ int* fd) noexcept
+{
+    _ebpf_operation_get_handle_by_id_request request{sizeof(request), operation, id};
+    _ebpf_operation_get_handle_by_id_reply reply;
+
+    uint32_t error = invoke_ioctl(request, reply);
+    ebpf_result_t result = windows_error_to_ebpf_result(error);
+    if (result != EBPF_SUCCESS) {
+        return result;
+    }
+
+    *fd = _get_next_file_descriptor((ebpf_handle_t)reply.handle);
```

Comment:
is it OK to return success if return value is ebpf_fd_invalid?

Reply:
Good point, updated